pmda/density.py

```diff
@@ -0,0 +1,158 @@
+            bin size for the density grid in Angstroem (same in x,y,z) [1.0]
+            Slice the trajectory as "trajectory[start:stop:step]"; default
+            is to read the whole trajectory.
+    metadata : dict. (optional)
+            `dict` of additional data to be saved with the object; the meta
+            data are passed through as they are.
+    padding : float (optional)
+            increase histogram dimensions by padding (on top of initial box
+            size) in Angstroem. Padding is ignored when setting a user defined
+            grid. [2.0]
+    updating : bool (optional)
+            Should the selection of atoms be updated for every step? ["False"]
+            - "True": atom selection is updated for each frame, can be slow
+            - "False": atoms are only selected at the beginning
+    parameters : dict (optional)
+            `dict` with some special parameters for :class:`Density` (see docs)
+    gridcenter : numpy ndarray, float32 (optional)
+            3 element numpy array detailing the x, y and z coordinates of the
+            center of a user defined grid box in Angstroem ["None"]
+    xdim : float (optional)
+            User defined x dimension box edge in ångström; ignored if
+            gridcenter is "None"
+    ydim : float (optional)
+            User defined y dimension box edge in ångström; ignored if
+            gridcenter is "None"
+    zdim : float (optional)
+            User defined z dimension box edge in ångström; ignored if
+            gridcenter is "None"
+    """
+    def __init__(self, atomgroup, delta=1.0, atomselection=None,
+                 metadata=None, padding=2.0, updating=False,
+                 parameters=None, gridcenter=None, xdim=None, ydim=None,
+                 zdim=None):
+        u = atomgroup.universe
+        super(DensityAnalysis, self).__init__(u, (atomgroup, ))
+        self._atomgroup = atomgroup
+        self._delta = delta
+        self._atomselection = atomselection
+        self._metadata = metadata
+        self._padding = padding
+        self._updating = updating
+        self._parameters = parameters
+        self._gridcenter = gridcenter
+        self._xdim = xdim
+        self._ydim = ydim
+        self._zdim = zdim
+        self._trajectory = u.trajectory
+        self._n_frames = u.trajectory.n_frames
+        if updating and atomselection is None:
+            raise ValueError("updating=True requires a atomselection string")
+        elif not updating and atomselection is not None:
+            raise ValueError("""With updating=False, the atomselection='{}' is
+                        not used and should be None""".format(atomselection))
+
+    def _prepare(self):
+        coord = self.current_coordinates(self._atomgroup, self._atomselection,
+                                         self._updating)
+        box, angles = (self._trajectory.ts.dimensions[:3],
+                       self._trajectory.ts.dimensions[3:])
```

Comment:
`Box` and `angles` are not used in other place, make sure whether we need them or not. (look at pylint error https://travis-ci.org/MDAnalysis/pmda/jobs/559215737)